plugin.video.ted.talks/resources/lib/ted_talks.py

```diff
@@ -0,0 +1,327 @@
+import urllib
+import ted_talks_scraper
+import plugin
+import settings
+from model.fetcher import Fetcher
+from model.rss_scraper import NewTalksRss
+from model.speakers_scraper import Speakers
+from model.search_scraper import Search
+from model.topics_scraper import Topics
+import menu_util
+import os
+import time
+import xbmc
+import xbmcplugin
+import xbmcgui
+import xbmcaddon
+import itertools
+import sys
+
+
+class UI:
+
+    def __init__(self, get_HTML, ted_talks):
+        self.get_HTML = get_HTML
+        self.ted_talks = ted_talks
+
+    def endofdirectory(self, sortMethod='title', updateListing=False):
+        # set sortmethod to something xbmc can use
+        if sortMethod == 'title':
+            sortMethod = xbmcplugin.SORT_METHOD_LABEL
+        elif sortMethod == 'date':
+            sortMethod = xbmcplugin.SORT_METHOD_DATE
+        elif sortMethod == 'none':
+            sortMethod = xbmcplugin.SORT_METHOD_NONE
+
+        # Sort methods are required in library mode.
+        xbmcplugin.addSortMethod(int(sys.argv[1]), sortMethod)
+        # let xbmc know the script is done adding items to the list.
+        xbmcplugin.endOfDirectory(handle=int(sys.argv[1]), updateListing=updateListing)
+
+    def addItem(self, title, mode, url=None, img='', args={}, video_info={}, isFolder=True, total_items=0):
+        # Create action url
+        args['mode'] = mode
+        if url:
+            args['url'] = url
+
+        args = [k + '=' + urllib.quote_plus(v.encode('ascii', 'ignore')) for k, v in args.iteritems()]
+        action_url = sys.argv[0] + '?' + "&".join(args)
+
+        li = xbmcgui.ListItem(label=title, iconImage=img, thumbnailImage=img)
```

Comment:
`iconImage` and `thumbnailImage` are deprecated, use `setArt(...)`

Reply:
Thanks for your feedback @Rechi. I've made the suggested improvements.